wallet/src/de/schildbach/wallet/data/TransactionMetadataDocumentDao.kt

```diff
@@ -37,60 +37,51 @@ interface TransactionMetadataDocumentDao {
     @Query("""
         SELECT sentTimestamp 
         FROM transaction_metadata_platform 
-        WHERE timestamp = (
-            SELECT MAX(timestamp) 
-            FROM transaction_metadata_platform 
-            WHERE txId = :txId
-                AND sentTimestamp is NOT NULL
-        )
-        """)
+        WHERE txId = :txId 
+            AND sentTimestamp is NOT NULL
+        ORDER BY timestamp DESC
+        LIMIT 1
+    """)
     suspend fun getSentTimestamp(txId: Sha256Hash): Long?
 
     @Query("""
         SELECT memo 
         FROM transaction_metadata_platform 
-        WHERE timestamp = (
-            SELECT MAX(timestamp) 
-            FROM transaction_metadata_platform 
-            WHERE txId = :txId
-                AND memo is NOT NULL
-        )
-        """)
+        WHERE txId = :txId 
+            AND memo is NOT NULL
+        ORDER BY timestamp DESC
+        LIMIT 1
+    """)
```

Comment:
The old query had a problem: it didn't filter by `txId` correctly because the filtering was only done inside the subquery, and the main query was returning any row that would match `WHERE timestamp =` condition. This caused several transactions to display the same memo.

The old query could also be modified like this, but the simplified version should be slightly more performant:

```
@Query("""
    SELECT memo 
    FROM transaction_metadata_platform 
    WHERE timestamp = (
        SELECT MAX(timestamp) 
        FROM transaction_metadata_platform 
        WHERE txId = :txId
            AND memo is NOT NULL
    ) AND txId = :txId
""")
```